opm/autodiff/MultisegmentWells.cpp

```diff
@@ -381,6 +385,55 @@ namespace Opm {
         return indices;
     }
 
+
+
+
+
+    WellCollection*
+    MultisegmentWells::
+    wellCollection() const {
+        return well_collection_;
+    }
+
+
+
+
+
+    void
+    MultisegmentWells::
+    calculateEfficiencyFactors()
+    {
+        if ( !localWellsActive() ) {
+            return;
+        }
+        // get efficiency factor for each well first
+        const int nw = wells_->number_of_wells;
+
+        Vector well_efficiency_factors = Vector::Ones(nw);
+
+        for (int w = 0; w < nw; ++w) {
+            const std::string well_name = wells_->name[w];
+            const WellNode* well_node = dynamic_cast<const WellNode *>(well_collection_->findNode(well_name));
```

Comment:
I think we can find a more robust solution for this cast. I'll think about it a little.